### PR TITLE
Mem optimize

### PR DESCRIPTION
**Pull request by ladisgin**
## Checklist:
- [ ] The PR addresses a single issue.  If it can be divided into multiple independent PRs, please do so.
- [ ] The PR is divided into a logical sequence of commits OR a single commit is sufficient.
- [ ] There are no unnecessary commits (e.g. commits fixing issues in a previous commit in the same PR).
- [ ] Each commit has a meaningful message documenting what it does.
- [ ] All messages added to the codebase, all comments, as well as commit messages are spellchecked.
- [ ] The code is commented OR not applicable/necessary.
- [ ] The patch is formatted via clang-format OR not applicable (if explicitly overridden leave unchecked and explain).
- [ ] There are test cases for the code you added or modified OR no such test cases are required.

### CODE REVIEW

File: include/klee/Module/KInstruction.h [hunk 1]
```diff
@@ -11,15 +11,16 @@
 #define KLEE_KINSTRUCTION_H
 
 #include "klee/Config/Version.h"
-#include "klee/Module/InstructionInfoTable.h"
 
 #include "klee/Support/CompilerWarning.h"
 DISABLE_WARNING_PUSH
 DISABLE_WARNING_DEPRECATED_DECLARATIONS
+#include "KModule.h"
```
Comment (misonijnik) [inline include/klee/Module/KInstruction.h:18]: Warnings should be disabled only for llvm-related headers.

File: include/klee/Module/KModule.h [hunk 2]
```diff
@@ -131,33 +126,33 @@ struct KReturnBlock : KBlock {
...
-  std::unordered_map<unsigned, KInstruction *> registerToInstructionMap;
-  unsigned numInstructions;
-  unsigned numBlocks;
   KInstruction **instructions;
 
-  bool kleeHandled = false;
+  [[nodiscard]] KInstruction *getInstructionByRegister(size_t reg) const;
 
   std::unordered_map<const llvm::Instruction *, KInstruction *> instructionMap;
+  // TODO maybe change to std::unique_ptr<KBlock[]>
```
Comment (misonijnik) [inline include/klee/Module/KModule.h:139]: Then you will need to clear the memory allocated for KBlocks.

File: lib/Module/KInstruction.cpp [hunk 3]
```diff
@@ -9,31 +9,110 @@
...
 
 #include "klee/Support/CompilerWarning.h"
 DISABLE_WARNING_PUSH
 DISABLE_WARNING_DEPRECATED_DECLARATIONS
+#include "klee/Support/ErrorHandling.h"
 #include "llvm/ADT/StringExtras.h"
+#include "llvm/IR/DebugInfoMetadata.h"
 DISABLE_WARNING_POP
 
+#include <llvm/IR/InstrTypes.h>
```
Comment (misonijnik) [inline lib/Module/KInstruction.cpp:22]: Move llvm-related includes into the disabling warning region.

File: lib/Module/KInstruction.cpp [hunk 4]
```diff
@@ -9,31 +9,110 @@
 
 #include "klee/Module/KInstruction.h"
 #include "klee/Module/KModule.h"
+#include "klee/Module/LocationInfo.h"
 
 #include "klee/Support/CompilerWarning.h"
 DISABLE_WARNING_PUSH
 DISABLE_WARNING_DEPRECATED_DECLARATIONS
+#include "klee/Support/ErrorHandling.h"
```
Comment (misonijnik) [inline lib/Module/KInstruction.cpp:17]: Move outside the disabling warning region.

File: lib/Module/KModule.cpp [hunk 5]
```diff
@@ -46,7 +48,10 @@ DISABLE_WARNING_DEPRECATED_DECLARATIONS
 #include "llvm/Transforms/Utils/Cloning.h"
 DISABLE_WARNING_POP
 
+#include <llvm/Support/FormattedStream.h>
```
Comment (misonijnik) [inline lib/Module/KModule.cpp:51]: Move into the disabling warning region.

**Review by misonijnik — CHANGES_REQUESTED** [review]

File: include/klee/Module/KInstruction.h (hunk 1 above)
Comment (ladisgin) [reply include/klee/Module/KInstruction.h]: Done

File: lib/Module/KInstruction.cpp (hunk 4 above)
Comment (ladisgin) [reply lib/Module/KInstruction.cpp]: Done

File: lib/Module/KModule.cpp (hunk 5 above)
Comment (ladisgin) [reply lib/Module/KModule.cpp]: Done

File: lib/Module/KInstruction.cpp (hunk 3 above)
Comment (ladisgin) [reply lib/Module/KInstruction.cpp:19]: Done

File: include/klee/Module/KModule.h (hunk 2 above)
Comment (ladisgin) [reply include/klee/Module/KModule.h]: Deleted this comment.
It was comment about ways to optimize, but this one doesn't get real profit.

**Review by misonijnik — APPROVED** [review]

**Pull request merged**